build/builder.go

```diff
@@ -689,7 +690,15 @@ func (b *Builder) Finish() error {
 	return b.buildError
 }
 
-func compareBranches(a, b []zoekt.RepositoryBranch) IndexState {
+// CompareBranches returns an IndexState comparing the two zoekt.RepositoryBranch
+// slices.
+//
+// The result will be IndexStateEqual if both slices specify the same set
+// of branch names and versions, IndexStateBranchSet if either slice specifies a different
+// set of branch names than the other, IndexStateBranchVersion if both slices specify the same set of
+// branch names but have different accompanying versions, or IndexStateCorrupt if one of the slices contains
+// an "invalid" branch set (such as duplicate branch names).
+func CompareBranches(a, b []zoekt.RepositoryBranch) IndexState {
 	set := make(map[string]string, len(a))
```

Comment:
another PR: you don't need to make this a set, you can (and should) rely on consistent ordering of branch names. This name order is significant both in how indexes are laid out at read time. Technically for comparison it doesn't matter, but in the past if the order changed we treated it as a change. So you can simplify code here.

Reply:
Will be resolved in https://github.com/sourcegraph/zoekt/pull/304 (thanks!)